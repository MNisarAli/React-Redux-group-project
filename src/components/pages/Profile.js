import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchedMissions } from '../../redux/missions/mission';
import '../../style.css';

const Profile = () => {
  const dispatch = useDispatch();
  const missionsData = useSelector((state) => state.missions);
  //   const { missionsStatus, missions } = missionsData;

  const missionsStatus = missionsData.status;
  const Mission = missionsData.missions;

  useEffect(() => {
    if (missionsStatus === 'idle') {
      dispatch(fetchedMissions());
    }
  }, [dispatch, missionsStatus]);
  return (
    <main>
      <div className="flex">
        <div>
          <h3>My Missions</h3>
          <div>
            {
              Mission.filter((mission) => (
                mission.isReserved === true
              )).map((mission) => (
                <div className="ele" key={mission.id}>
                  <h4>{mission.name}</h4>
                </div>
              ))
            }
          </div>
        </div>
        <div>
          <h3>My Rockets</h3>
          <div />
        </div>
      </div>
    </main>

  );
};

export default Profile;
