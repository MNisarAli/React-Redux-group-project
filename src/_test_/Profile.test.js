import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Profile from '../components/pages/Profile';
import store from '../redux/configureStore';

it('matches snapshot', () => {
  const tree = renderer.create(
    <Provider store={store}><Profile /></Provider>,
  )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
