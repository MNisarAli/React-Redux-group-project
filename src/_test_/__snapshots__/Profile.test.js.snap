// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="reserved-container"
>
  <article>
    <h2>
      My Missions
    </h2>
    <ul
      className="reserved-list"
    />
  </article>
  <article>
    <h2>
      My Rockets
    </h2>
    <ul
      className="reserved-list"
    />
  </article>
</section>
`;
